packages/type-storage/src/fromMetadata/createFunction.spec.ts

```diff
@@ -53,4 +55,60 @@ describe('createFunction', () => {
       )
     );
   });
+
+  describe('the created double map function', () => {
+    let storageFn: StorageFunction;
+    beforeAll(() => {
+      storageFn = createFunction(
+        'GenericAsset',
+        'FreeBalance', {
+          name: 'metaName',
+          type: {
+            isDoubleMap: true,
+            asDoubleMap: {
+              hasher: new StorageHasher('Blake2_256'),
+              key1: new Text('AccountId'),
+              key2: new Text('AccountId'),
+              value: new Text('Balance'),
+              key2Hasher: new Text('twox_128')
+            }
+          }
+        } as any
+      );
+    });
+
+    it('should return correct key', () => {
+      const result = storageFn(['5DXUeE5N5LtkW97F2PzqYPyqNkxqSWESdGSPTX6AvkUAhwKP', '5DXUeE5N5LtkW97F2PzqYPyqNkxqSWESdGSPTX6AvkUAhwKP']);
+      expect(u8aToHex(result)).toEqual('0x00fa40e72d7173e69ee54b980345ea01cb81e64258502e0247af4303dee91ec0aec2ecd3a60ab080cff7b52a8f6d543b');
+    });
```

Comment:
This expected value '0x00fa40e72d7173e69ee54b980345ea01cb81e64258502e0247af4303dee91ec0aec2ecd3a60ab080cff7b52a8f6d543b' here is from the intergration test I did on our chain. This value is the key on the chain. And the storagefunction created here returns the same key.